src/k8s/pkg/k8sd/pki/etcd.go

```diff
@@ -0,0 +1,137 @@
+package pki
+
+import (
+	"crypto/x509/pkix"
+	"fmt"
+	"net"
+)
+
+// EtcdPKI is a list of certificates required by the etcd datastore.
+type EtcdPKI struct {
+	allowSelfSignedCA bool     // create self-signed CA certificates if missing
+	hostname          string   // node name
+	ipSANs            []net.IP // IP SANs for generated certificates
+	dnsSANs           []string // DNS SANs for the certificates below
+	years             int      // how many years the generated certificates will be valid for
+
+	// CN=k8s-dqlite, DNS=hostname, IP=127.0.0.1 (self-signed)
+	CACert, CAKey string
+
+	// [server] CN=kube-etcd, DNS=hostname, IP=127.0.0.1,address (signed by etcd-ca)
+	ServerCert, ServerKey string
+
+	// [server] CN=kube-etcd-peer, DNS=hostname, IP=127.0.0.1,address (signed by etcd-ca)
+	ServerPeerCert, ServerPeerKey string
+
+	// [client] CN=kube-apiserver-etcd-client (signed by etcd-ca)
+	APIServerClientCert, APIServerClientKey string
+}
+
+type EtcdPKIOpts struct {
+	Hostname          string
+	DNSSANs           []string
+	IPSANs            []net.IP
+	Years             int
+	AllowSelfSignedCA bool
+}
+
+func NewEtcdPKI(opts EtcdPKIOpts) *EtcdPKI {
+	if opts.Years == 0 {
+		opts.Years = 10
```

Comment:
What happens when the self-signed CA expires? Do you have to manually create new ones?